src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java

```diff
@@ -34,13 +36,36 @@ public String getUrlName() {
     }
 
     public final Object doRedirect(StaplerRequest req, StaplerResponse rsp) throws IOException {
-        DisplayURLProvider provider = lookupProvider();
+        DisplayURLProvider provider = lookupProvider(req);
         rsp.sendRedirect(HttpServletResponse.SC_MOVED_TEMPORARILY, getRedirectURL(provider));
         return null;
     }
 
     protected abstract String getRedirectURL(DisplayURLProvider provider);
 
+    DisplayURLProvider lookupProvider(StaplerRequest req) {
+        final String providerName = req.getParameter("provider");
+        if(providerName != null && !providerName.isEmpty()) {
+            Iterable<DisplayURLProvider> providers = DisplayURLProvider.all();
+            Iterable<DisplayURLProvider> filtered = Iterables.filter(providers, new com.google.common.base.Predicate<DisplayURLProvider>() {
```

Comment:
nit: import Predicate

Reply:
oops, i had imported java predicate and then intellij did the above